### PR TITLE
Temp disable MKL in  DistributionKernels.cpp

### PR DESCRIPTION
**Pull request by malfet**
Until https://github.com/pytorch/pytorch/issues/132395 is addressed

Test plan: Add test based on the script below (taken from https://discuss.pytorch.org/t/bug-in-torch-multinomial-generated-distribution-is-modestly-incorrect-edit-this-is-a-regression-and-appears-to-be-due-to-an-analogous-bug-in-tensor-exponential )
```python
import torch

high_bits_for_seed = 16000000000000000000           # to use "good quality" seed
_ = torch.manual_seed (high_bits_for_seed + 2024)

prob = torch.ones (26)
dups_mult = 0
perm_counts_mult = {}
for _ in range (1_000_000):
    p = tuple (torch.multinomial (prob, prob.numel(), replacement=False).tolist())
    if  p in perm_counts_mult:
        dups_mult += 1
        perm_counts_mult[p] += 1
    else:
        perm_counts_mult[p] = 1

print ('duplicate multinomial perms: ', dups_mult)
print ('multiple multinomial perms:  ', (torch.tensor (list (perm_counts_mult.values())) > 1).sum().item())
print ('max of perm_counts_mult:     ', torch.tensor (list (perm_counts_mult.values())).max().item())
print ('len (perm_counts_mult):      ', len (perm_counts_mult))
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
Sounds like a good temporary fix.
Could we add the example as a slowtest to make sure we don't regress again?

File: test/distributions/test_distributions.py [hunk 1]
```diff
@@ -1841,6 +1841,21 @@ def test_multinomial_2d(self):
             torch.tensor([[total_count, 0], [0, total_count]], dtype=torch.float64),
         )
 
+    def test_multinomial_sequential_draw(self):
+        # Adapted after script mentioned in https://github.com/pytorch/pytorch/issues/132395
+        torch.manual_seed(0xde0b6b3a764007e8)
```
Comment (malfet) [inline test/distributions/test_distributions.py:1846]:
```suggestion
        torch.manual_seed(0xDE0B6B3A764007E8)
```

**Review by albanD — APPROVED** [review]
Thanks!

**Comment by malfet** [comment]
@PaliC is there a way you can share some info about those instabilities publically? Because change fixes a pretty big problem in randomness, which makes me wonder if those tests internally are wrong

**Pull request closed without merging**